Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3489,55 +3492,40 @@ - (BOOL)shouldAnimateRowUpdates:(NSArray<YapDatabaseViewRowChange *> *)rowChange
         switch (rowChange.type) {
             case YapDatabaseViewChangeDelete:
-                isOnlyInsertingNewOutgoingMessages = NO;
-                isOnlyUpdatingLastOutgoingMessage = NO;
-                if (!lastNonUpdateRow || lastNonUpdateRow.integerValue < rowChange.indexPath.row) {
-                    lastNonUpdateRow = @(rowChange.indexPath.row);
-                }
+                isOnlyModifyingLastMessage = NO;
                 break;
             case YapDatabaseViewChangeInsert: {
-                isOnlyUpdatingLastOutgoingMessage = NO;
                 ConversationViewItem *_Nullable viewItem = [self viewItemForIndex:(NSInteger)rowChange.finalIndex];
-                if ([viewItem.interaction isKindOfClass:[TSOutgoingMessage class]]
+                if (([viewItem.interaction isKindOfClass:[TSIncomingMessage class]] ||
+                        [viewItem.interaction isKindOfClass:[TSOutgoingMessage class]])
                     && rowChange.finalIndex >= oldViewItemCount) {
                     continue;
                 }
-                if (!lastNonUpdateRow || lastNonUpdateRow.unsignedIntegerValue < rowChange.finalIndex) {
-                    lastNonUpdateRow = @(rowChange.finalIndex);
-                }
+                isOnlyModifyingLastMessage = NO;
             }
             case YapDatabaseViewChangeMove:
-                isOnlyInsertingNewOutgoingMessages = NO;
-                isOnlyUpdatingLastOutgoingMessage = NO;
-                if (!lastNonUpdateRow || lastNonUpdateRow.integerValue < rowChange.indexPath.row) {
-                    lastNonUpdateRow = @(rowChange.indexPath.row);
-                }
-                if (!lastNonUpdateRow || lastNonUpdateRow.unsignedIntegerValue < rowChange.finalIndex) {
-                    lastNonUpdateRow = @(rowChange.finalIndex);
-                }
+                isOnlyModifyingLastMessage = NO;
                 break;
             case YapDatabaseViewChangeUpdate: {
-                isOnlyInsertingNewOutgoingMessages = NO;
-                ConversationViewItem *_Nullable viewItem = [self viewItemForIndex:(NSInteger)rowChange.finalIndex];
-                if (![viewItem.interaction isKindOfClass:[TSOutgoingMessage class]]
-                    || rowChange.indexPath.row != (NSInteger)(oldViewItemCount - 1)) {
-                    isOnlyUpdatingLastOutgoingMessage = NO;
+                if (rowChange.changes == YapDatabaseViewChangedDependency) {
+                    continue;
                 }
-                if (!lastUpdateRow || lastUpdateRow.integerValue < rowChange.indexPath.row) {
-                    lastUpdateRow = @(rowChange.indexPath.row);
+                ConversationViewItem *_Nullable viewItem = [self viewItemForIndex:(NSInteger)rowChange.finalIndex];
+                if (([viewItem.interaction isKindOfClass:[TSIncomingMessage class]] ||
+                        [viewItem.interaction isKindOfClass:[TSOutgoingMessage class]])
```

Comment:
We want new incoming _and_ outgoing messages to feel snappy.

Reply:
This is the only part of these changes that feels controversial to me - maybe we should animate in incoming messages.